src/helics/apps/Connector.cpp

```diff
@@ -0,0 +1,1202 @@
+    auto iview2 = addInterface(interface2);
+    Connection conn{iview1, iview2, direction, std::move(svtags)};
+    if (iview1.compare(0, 6, "REGEX:") == 0) {
+        switch (direction) {
+            case InterfaceDirection::TO_FROM:
+                std::swap(conn.interface1, conn.interface2);
+                matchers.emplace_back(std::move(conn));
+                break;
+            case InterfaceDirection::FROM_TO:
+                matchers.emplace_back(std::move(conn));
+                break;
+            case InterfaceDirection::BIDIRECTIONAL:
+                matchers.emplace_back(conn);
+                std::swap(conn.interface1, conn.interface2);
+                matchers.emplace_back(std::move(conn));
+                break;
+        }
+    } else {
+        switch (direction) {
+            case InterfaceDirection::TO_FROM:
+                connections.emplace(iview2, std::move(conn));
+                break;
+            case InterfaceDirection::FROM_TO:
+                connections.emplace(iview1, std::move(conn));
+                break;
+            case InterfaceDirection::BIDIRECTIONAL:
+                connections.emplace(iview2, conn);
+                if (iview1 != iview2) {
+                    connections.emplace(iview1, std::move(conn));
+                }
+                break;
+        }
+    }
+}
+
+class RegexMatcher {
+  public:
+    RegexMatcher() = default;
+
+    std::regex rmatch;
+    std::vector<std::string> keys;
+    std::string_view interface1;
+    std::string_view interface2;
+    std::vector<std::size_t> tags;
+    std::string generateMatch(std::string_view testString)
+    {
+        std::match_results<typename decltype(testString)::const_iterator> matchResults{};
+        if (std::regex_match(testString.begin(), testString.end(), matchResults, rmatch)) {
+            std::string matcher(interface2);
+            if (matcher.compare(0, 6, "REGEX:") == 0) {
+                matcher.erase(0, 6);
+                for (std::size_t ii = 0; ii < keys.size(); ++ii) {
+                    auto keyloc = matcher.find(keys[ii]);
+                    while (keyloc != std::string::npos) {
+                        auto endloc = matcher.find_first_of(')', keyloc);
+                        matcher.replace(matcher.begin() + keyloc - 1,
+                                        matcher.begin() + endloc + 1,
+                                        matchResults[ii + 1].first,
+                                        matchResults[ii + 1].second);
+                        keyloc = matcher.find(keys[ii]);
+                    }
+                }
+            }
+
+            return matcher;
+        }
+        return {};
+    }
+};
+
+void Connector::loadTextFile(const std::string& filename)
+{
+    App::loadTextFile(filename);
+    using namespace gmlc::utilities::stringOps;  // NOLINT
+    std::ifstream infile(filename);
+    std::string str;
+
+    int ccnt = 0;
+    bool mlineComment = false;
```

Comment:
I think this could use a comment saying that it is for multi-line comments.

Alternatively, it might make sense to just remove the multi-line comment feature; it looks like it uses a syntax that I don't think is commonly used, and the file format is simple enough that it may be an unnecessary feature (Python doesn't have an official multi-line comment syntax either).